lib/endpoints/class-wp-rest-posts-controller.php

```diff
@@ -240,7 +240,10 @@ public function create_item_permissions_check( $request ) {
 			return new WP_Error( 'rest_cannot_assign_sticky', __( 'You do not have permission to make posts sticky.' ), array( 'status' => rest_authorization_required_code() ) );
 		}
 
-		return current_user_can( $post_type->cap->create_posts );
+		if ( ! current_user_can( $post_type->cap->create_posts ) ) {
+			return new WP_Error( 'rest_cannot_create', __( 'Sorry, you are not allowed to create new posts.' ), array( 'status' => rest_authorization_required_code() ) );
```

Comment:
Should we be using the post type labels here? Trying to create a page will show "posts" right now